apps/core/system/ui/runtime.py

```
@@ -0,0 +1,138 @@
+                continue
+            if 1 <= port <= 65535:
+                return port
+    return None
+
+
+def _detect_runserver_process() -> tuple[bool, int | None]:
+    """Return whether the dev server is running and the port if available."""
+
+    try:
+        result = subprocess.run(
+            ["pgrep", "-af", "manage.py runserver"],
```

Ruff (0.14.14)

[error] Starting a process with a partial executable path

(S607)